### DISCUSSION TITLE
403 on VerifyEmail when the link has timed out

### DISCUSSION THREAD

**Discussion by markdynamo6** [filamentphp/filament#11435] [Help] [answered]
### Package

Notifications

### Package Version

3.2.2

### How can we help you?

When the verify email link has expired (as defined by **auth.verification.expire**, I get a **403 | invalid signiture** page. Is this expected behaviour?  I'm expecting to see the _auth.email-verification.prompt_ with a message to say that the link has expired, and a link to resend.

Or have I just mis-configured something? Some people are suggesting adding the URI to the VerifyCsrfToken exception list, but I don't want to blindly do that if there is a Filament solution.

Note that If the link has not expired then the email verification works as expected.

**Comment by leandrocfe** [comment, marked as answer]
Another way would be customizing the handler method in the `app/Http/Middleware/ValidateSignature.php` middleware:

```php
public function handle($request, Closure $next, ...$args)
{
    [$relative, $ignore] = $this->parseArguments($args);

    if ($request->hasValidSignatureWhileIgnoring($ignore, ! $relative)) {
        return $next($request);
    }

    if (str_contains($request->getPathInfo(), 'email-verification')) {
        //redirect the user to the email verification page or something
    }

    throw new InvalidSignatureException;
}
```

**Reply by markdynamo6** [reply]
Cheers Leandro
